packages/contracts-rfq/contracts/FastBridgeV2.sol

```diff
@@ -183,31 +182,34 @@ contract FastBridgeV2 is Admin, IFastBridgeV2 {
     function prove(bytes memory request, bytes32 destTxHash, address relayer) public onlyRole(RELAYER_ROLE) {
         bytes32 transactionId = keccak256(request);
         // update bridge tx status given proof provided
-        if (bridgeStatuses[transactionId] != BridgeStatus.REQUESTED) revert StatusIncorrect();
-        bridgeStatuses[transactionId] = BridgeStatus.RELAYER_PROVED;
-        bridgeProofs[transactionId] = BridgeProof({timestamp: uint96(block.timestamp), relayer: relayer}); // overflow ok
+        if (bridgeTxDetails[transactionId].status != BridgeStatus.REQUESTED) revert StatusIncorrect();
+        BridgeTxDetails storage _thisBridgeTxDetails = bridgeTxDetails[transactionId];
+        _thisBridgeTxDetails.status = BridgeStatus.RELAYER_PROVED;
+        _thisBridgeTxDetails.proofBlockTimestamp = uint40(block.timestamp);
+        _thisBridgeTxDetails.proofBlockNumber = uint48(block.number);
+        _thisBridgeTxDetails.proofRelayer = relayer;
 
         emit BridgeProofProvided(transactionId, relayer, destTxHash);
     }
 
     /// @notice Calculates time since proof submitted
-    /// @dev proof.timestamp stores casted uint96(block.timestamp) block timestamps for gas optimization
-    ///      _timeSince(proof) can accomodate rollover case when block.timestamp > type(uint96).max but
-    ///      proof.timestamp < type(uint96).max via unchecked statement
-    /// @param proof The bridge proof
+    /// @dev proof.timestamp stores casted uint40(block.timestamp) block timestamps for gas optimization
+    ///      _timeSince(proof) can accomodate rollover case when block.timestamp > type(uint40).max but
+    ///      proof.timestamp < type(uint40).max via unchecked statement
+    /// @param proofBlockTimestamp The bridge proof block timestamp
     /// @return delta Time delta since proof submitted
-    function _timeSince(BridgeProof memory proof) internal view returns (uint256 delta) {
+    function _timeSince(uint40 proofBlockTimestamp) internal view returns (uint256 delta) {
         unchecked {
-            delta = uint96(block.timestamp) - proof.timestamp;
+            delta = uint40(block.timestamp) - proofBlockTimestamp;
         }
     }
 
     /// @inheritdoc IFastBridge
     function canClaim(bytes32 transactionId, address relayer) external view returns (bool) {
-        if (bridgeStatuses[transactionId] != BridgeStatus.RELAYER_PROVED) revert StatusIncorrect();
-        BridgeProof memory proof = bridgeProofs[transactionId];
-        if (proof.relayer != relayer) revert SenderIncorrect();
-        return _timeSince(proof) > DISPUTE_PERIOD;
+        BridgeTxDetails storage _thisBridgeTxDetails = bridgeTxDetails[transactionId];
+        if (_thisBridgeTxDetails.status != BridgeStatus.RELAYER_PROVED) revert StatusIncorrect();
+        if (_thisBridgeTxDetails.proofRelayer != relayer) revert SenderIncorrect();
+        return _timeSince(_thisBridgeTxDetails.proofBlockTimestamp) > DISPUTE_PERIOD;
```

Comment:
Do we also need storage approach here? What if we do something like this instead?

`(BridgeStatus status, uint40 proofBlockTimestamp, , address proofRelayer) = bridgeTxDetails[transactionId]`